### PR TITLE
Add Cutelyst framework tests

### PR DESCRIPTION
**Pull request by dantti**
This includes all 6 test as
well as json configuration for
running them. Since Cutelyst has
a pluggable Engine which the best one is
uwsgi, the test use it to process the
HTTP requests directly, as well as
Nginx as a front-end giving it
keep-alive support is also available.

### CODE REVIEW

**Comment by dantti** [comment]
@nbrady-techempower will round-14 be based on 14.04? if so I need to add a cmake ppa as it has a too old cmake, in fact most of stuff there was old on release time....

**Comment by nbrady-techempower** [comment]
@dantti having a problem here:
```bash
INFO:root:Executed setup.sh, returning 1
Server Cutelyst: ++++ command='sudo apt-add-repository --yes ppa:beineri/opt-qt$QT_VERSION_FULL'
Server Cutelyst: ++++ IFS=:
Server Cutelyst: ++++ read -a funcstack
Server Cutelyst: ++++ IFS=:
Server Cutelyst: ++++ read -a bashstack
Server Cutelyst: ++++ IFS=:
Server Cutelyst: ++++ read -a linestack
Server Cutelyst: ++++ FW_dep_error=1
Server Cutelyst: ++++ FW_any_errors=1
Server Cutelyst: +++++ pwd
Server Cutelyst: ++++ wd=/home/vagrant/FrameworkBenchmarks/installs
Server Cutelyst: ++++ relative_wd='$FWROOT/installs'
Server Cutelyst: +++++ echo /toolset/setup/linux/frameworks/cutelyst.sh
Server Cutelyst: ++++ echo 'ERROR: /toolset/setup/linux/frameworks/cutelyst.sh: Command '\''sudo apt-add-repository --yes ppa:beineri/opt-qt$QT_VERSION_FULL'\'' exited with status 1 (dependency=cutelyst) (cwd=$FWROOT/installs)'
Server Cutelyst: ERROR: /toolset/setup/linux/frameworks/cutelyst.sh: Command 'sudo apt-add-repository --yes ppa:beineri/opt-qt$QT_VERSION_FULL' exited with status 1 (dependency=cutelyst) (cwd=$FWROOT/installs)
Server Cutelyst: Framework processes have terminated
```

**Comment by nbrady-techempower** [comment]
@dantti ah yes, that is the current plan. feel free to make a file in systools for those needs, as we're planning on moving gcc, clang, etc there and then use fw_depends for it in your setup.sh

**Comment by dantti** [comment]
@nbrady-techempower Ok, I just forced an update for the Qt and CMake install fix, as well as updates to the changes in the API, should all be fine now

File: frameworks/C++/cutelyst/benchmark_config.json [hunk 1]
```diff
@@ -0,0 +1,127 @@
+{
+    "framework": "Cutelyst",
```
Comment (nbrady-techempower) [inline frameworks/C++/cutelyst/benchmark_config.json]: I would just change this to "cutelyst"

**Comment by nbrady-techempower** [comment]
Default Passes! I just made a note about changing the name to all lowercase. Running the other tests now.

```bash
Cutelyst: setup.sh and framework processes have terminated
Server Cutelyst: Framework processes have terminated
verify: 100% |####################################################################| Rough Time: 00:04:02
================================================================================
  Verification Summary

| Test: Cutelyst
|       plaintext   : PASS
|       json        : PASS
================================================================================

Time to complete: 243 seconds
Results are saved in results/20160621195330
```

**Comment by dantti** [comment]
@nbrady-techempower should I change all occurrences of Cutelyst to cutelyst in json file?

**Comment by nbrady-techempower** [comment]
@dantti That's probably best. It's the current convention we're going with

**Comment by dantti** [comment]
Done

**Comment by nbrady-techempower** [comment]
@dantti all tests passed except for nginx-postgres-raw

[log here](http://sprunge.us/YdKX)

**Comment by dantti** [comment]
@nbrady-techempower hmm the problem is that the DBHOST host is not set, so peer authentication fails, can I assume that if empty it is localhost? Dunno how it works on real tests but unix socket would probably be better performance wise

**Comment by dantti** [comment]
ok, postgres-raw test passes so something else is broken

**Comment by dantti** [comment]
@nbrady-techempower fixed!

**Comment by nbrady-techempower** [comment]
@dantti great! all tests pass!

**Comment by dantti** [comment]
@nbrady-techempower I'm thinking on adding more variants (ORM and template engines) to the test, are you going to merge this soon or can I keep force pushing to this PR?

**Comment by nbrady-techempower** [comment]
@dantti We don't merge the pr's until travis finishes and that takes some time usually, so you're welcome to push changes here and it will cancel the current travis run and add a new one to the queue. Plenty of time in round-14. If you don't want it to be merged just put the PR: Work in Progress tag on it

**Comment by nbrady-techempower** [comment]
@dantti I was curious what your status was with this? Would you like us to merge this in and then you can just open a new PR if you'd like to do more work on it?

**Comment by dantti** [comment]
@nbrady-techempower please merge, so that I don't miss some timeline, I'm already sad I couldn't make it for round-13... I'm not sure when I'll have time to write more tests variants. Thanks

**Pull request merged**